app/Http/Controllers/TarnibController.php

```diff
@@ -1,133 +1,182 @@
+            $room_id = $this->generateRandomString(); //todo make sure this is unique in db        
+
+            $id = DB::table('rooms')->insertGetId([
+                'room_id' => $room_id,
+                'player_1' => $request->player_name,
+                'player_1_token' => $request->player_token,
+                'team_1_score' => 0,
+                'team_2_score' => 0,
+            ]);
+
+            DB::table('rooms')->where('id', $id)->update([
+                'round_id' => $this->createRound($id)
+            ]);
 
-    function createRound($room_id){
-
-        $random_cards = $this->cards;
-        shuffle($random_cards);
-        $round_number = count(DB::table('rounds')->where('room_id', $room_id)->get());
-        $dealer= $round_number == 0 ? rand(0,3) : $round_number%4;
-        $round_id = DB::table('rounds')->insertGetId([
-            'room_id' => $room_id,
-            'player_1_cards' => json_encode($this->sortCards(array_slice($random_cards, 0, 13))),
-            'player_2_cards' => json_encode($this->sortCards(array_slice($random_cards, 13, 13))),
-            'player_3_cards' => json_encode($this->sortCards(array_slice($random_cards, 26, 13))),
-            'player_4_cards' => json_encode($this->sortCards(array_slice($random_cards, 39, 13))),
-            'turn' => 1,
-            'team_1_score' => 0,
-            'team_2_score' => 0,
-            'dealer' => $dealer,
-            'bids_data' => json_encode([
-                'bids' => [0,0,0,0],
-                'current_bidder' => ($round_number+1)%4
-            ]),
-            'current_play' => json_encode(['','','','']),
-            'player_turn' => -1,
-
-        ]);
-
-        return $round_id;
-    }
 
-    function joinRoom(Request $request){
-        $room = DB::table('rooms')->where('room_id', $request->room_id)->first();
-        $player_seat = '';
-        if($room->player_1 == $request->player_name) $player_seat = 1;
-        else if($room->player_2 == $request->player_name) $player_seat = 2;
-        else if($room->player_3 == $request->player_name) $player_seat = 3;
-        else if($room->player_4 == $request->player_name) $player_seat = 4;
+            return response()->json(['success' => true, 'room_id' => $room_id, 'player_name' => $request->player_name]);
 
+        }
 
-        $round_number = count(DB::table('rounds')->where('room_id', $room_id)->get());
+        function createRound($room_id){
+
+            $random_cards = $this->cards;
+            shuffle($random_cards);
+            $round_number = count(DB::table('rounds')->where('room_id', $room_id)->get());
+            $dealer= $round_number == 0 ? rand(0,3) : $round_number%4;
+            $round_id = DB::table('rounds')->insertGetId([
+                'room_id' => $room_id,
+                'player_1_cards' => json_encode($this->sortCards(array_slice($random_cards, 0, 13))),
+                'player_2_cards' => json_encode($this->sortCards(array_slice($random_cards, 13, 13))),
```

Comment:
seems like `array_slice` is used each time in the parameters. Why not just do that operation in the function itself